pennylane/spin/lattice.py

```diff
@@ -190,7 +206,76 @@ def _generate_grid(self, neighbour_order):
             lattice_points.append(point)
             lattice_map.append(node_index)
 
-        return math.array(lattice_points), math.array(lattice_map)
+        return math.array(lattice_points), lattice_map
+
+    def get_custom_edges(self, custom_edges, lattice_map):
+        """Generates the edges described in `custom_edges` for all unit cells.
+
+        **Example**
+
+        Generates a square lattice with a single diagonal and assigns a different operation
+        to horizontal, vertical, and diagonal edges.
+        >>> n_cells = [3,3]
+        >>> vectors = [[1, 0], [0,1]]
+        >>> custom_edges = [
+                [(0, 1), ("XX", 0.1)],
+                [(0, 3), ("YY", 0.2)],
+                [(0, 4), ("XY", 0.3)],
+                ]
+        >>> lattice = Lattice(n_cells=n_cells, vectors=vectors, custom_edges=custom_edges)
+
+        """
+
+        if not all([len(edge) in (1, 2) for edge in custom_edges]):
+            raise TypeError(
+                """
+                The elements of custom_edges should be lists of length 1 or 2. Inside said lists should be a tuple that contains two lattice indices to represent the edge and, optionally, a tuple that represents the operation and coefficient for that edge.
+                Every tuple must contain two lattice indices to represent the edge
+                and can optionally include a list to represent the operation and coefficient for that edge.
+                """
+            )
+
+        edges = []
+        n_sl = len(self.positions)
+        nsites_axis = math.cumprod([n_sl, *self.n_cells[:0:-1]])[::-1]
+
+        for i, custom_edge in enumerate(custom_edges):
+            edge = custom_edge[0]
+
+            if edge[0] >= self.n_sites or edge[1] >= self.n_sites:
+                raise ValueError(
+                    f"The edge {edge} has vertices greater than n_sites, {self.n_sites}"
+                )
+
+            edge_operation = custom_edge[1] if len(custom_edge) == 2 else i
+            map_edge1 = lattice_map.index(edge[0])
+            map_edge2 = lattice_map.index(edge[1])
+            edge_distance = self.lattice_points[map_edge2] - self.lattice_points[map_edge1]
+            v1, v2 = math.mod(edge, n_sl)
+            translation_vector = (
+                edge_distance + self.positions[v1] - self.positions[v2]
+            ) @ math.linalg.inv(self.vectors)
+            translation_vector = math.asarray(math.rint(translation_vector), dtype=int)
+            edge_ranges = []
+            for idx, cell in enumerate(self.n_cells):
+                if self.boundary_condition[idx]:
+                    edge_ranges.append(range(0, cell))
+                else:
+                    edge_ranges.append(
+                        range(
+                            math.maximum(0, -translation_vector[idx]),
+                            cell - math.maximum(0, translation_vector[idx]),
+                        )
+                    )
```

Comment:
```suggestion
            for idx, cell in enumerate(self.n_cells):
                t_point = 0 if self.boundary_condition[idx] else translation_vector[idx]
                edge_ranges.append(
                    range(math.maximum(0, -t_point), cell - math.maximum(0, t_point))
                )
```
this might be cleaner? 🤔